Assets/Scripts/Board/Board.cs

```diff
@@ -216,7 +224,96 @@ private void UpdateBoardUI()
+            {
+                if (graph.HasTile(neighbors[vertexMapping.Key]))
+                {
+                    Graph.TileNode neighbor = graph.GetTile(neighbors[vertexMapping.Key]);
+                    foreach (var map in vertexMapping.Value)
+                    {
+                        tileNode.AddVertex(map.Item1, neighbor.GetVertex(map.Item2));
+                    }
+                }
+            }
+            // else, create new vertex
+            foreach (Graph.VertexDirection direction in Enum.GetValues(typeof(Graph.VertexDirection)))
+            {
+                if (!tileNode.HasVertex(direction))
+                {
+                    Vector3 vertexWorldPosition = tileWorldPosition + Graph.GetVertexOffsetVector(direction);
+                    Graph.VertexNode vertexNode = new(vertexWorldPosition);
+                    Debug.Log("Created " + direction + " VertexNode at: " + vertexWorldPosition);
+
+                    foreach (Graph.VertexDirection neighborVertexDirection in Graph.GetVertexDirectionNeighbors(direction))
+                    {
+                        if (tileNode.HasVertex(neighborVertexDirection))
+                        {
+                            Graph.VertexNode neighborVertexNode = tileNode.GetVertex(neighborVertexDirection);
+
+                            Vector3 edgeWorldPosition = (vertexNode.GetWorldPosition() + neighborVertexNode.GetWorldPosition()) / 2;
+                            Graph.EdgeNode edgeNode = new(edgeWorldPosition);
+                            Debug.Log("Created EdgeNode at: " + edgeWorldPosition);
+
+                            edgeNode.AddVertex(vertexNode);
+                            edgeNode.AddVertex(neighborVertexNode);
+
+                            vertexNode.AddEdge(edgeNode);
+                            neighborVertexNode.AddEdge(edgeNode);
+
+                            graph.AddEdge(edgeNode);
+                        }
+                    }
+                    tileNode.AddVertex(direction, vertexNode);
+                    graph.AddVertex(vertexNode);
+                }
+            }
 
+            graph.AddTile(tile.position, tileNode);
+        }
+
+        Debug.Log("# TileNodes: " + graph.GetTiles().Count);
+        Debug.Log("# VertexNodes: " + graph.GetVertices().Count);
+        Debug.Log("# EdgeNodes: " + graph.GetEdges().Count);
+    }
+
+    private void TestGraphPoints()
+    {
+        foreach (Graph.TileNode tile in graph.GetTiles())
+        {
+            Instantiate(tileObject, tile.GetWorldPosition(), Quaternion.identity, graphLayer);
+        }
+        foreach (Graph.VertexNode vertex in graph.GetVertices())
+        {
+            Instantiate(vertexObject, vertex.GetWorldPosition(), Quaternion.identity, graphLayer);
+        }
+        foreach (Graph.EdgeNode edge in graph.GetEdges())
+        {
+            Instantiate(edgeObject, edge.GetWorldPosition(), Quaternion.identity, graphLayer);
+        }
```

Comment:
Can remove this once ready to submit, but also may want to keep the real version around and/or split into `AddPlacePoints` for edges/vertices

Reply:
SG